src/ppx/ppx_eliom_utils.ml

```diff
@@ -495,101 +501,242 @@ module Rpc = struct
-              ~%(Eliom_client.server_function
-                   ~name:
-                     [%e
-                       Exp.constant
-                         (Pconst_string (rpc_name fun_name_pattern, None))]
-                   [%json: [%t _argstyp]]
-                   (Os_session.connected_wrapper (fun [%p _args] ->
-                      [%e
-                        Exp.apply
-                          (Exp.ident
-                             (Location.mkloc
-                                (Longident.parse
-                                     (function_name fun_name_pattern))
-                                fun_name_pattern.ppat_loc))
-                          (List.map
-                             (fun (pattern, _, label) ->
-                                (label, pat_to_exp pattern))
-                             args_list)])))
-            in
-            [%e
-              Exp.apply
-                (Exp.ident (Location.mkloc (Longident.parse "aux") loc))
-                [(Asttypes.Nolabel, _formatargs)]]] [@metaloc loc] )
+          Exp.fun_ (Asttypes.Optional s) None pattern (expr_mapper expr')
+      | _ ->
+          let rpc_name =
+            Exp.constant (Pconst_string (rpc_name fun_name_pattern, None))
+          in
+          let fun_name = function_name fun_name_pattern in
+          let aux_apply =
+            apply_function_expr ~loc "aux" [(Asttypes.Nolabel, _expr_of_args)]
+          in
+          apply_rpc ~loc rpc_name fun_name args_list _pat_of_args _typ_of_args aux_apply
     in
     expr_mapper expr
 
-  let client_structure_stri stri =
+  let server_side rpc_type stri =
     let loc = stri.pstr_loc in
     match stri with
-    | [%stri let [%p? pattern] = [%e? expr]] -> (
-        ([ [%stri let [%p pattern] = [%e client_expr pattern expr]]
-         ] [@metaloc loc]) )
+    | [%stri
+        let [%p? {ppat_desc= Ppat_var ident} as pattern] =
+          [%e? [%expr fun [%p? _] -> [%e? expr']]]] ->
+        let args_list = args_parser expr' [] in
+        let _args =
+          List.map (fun (pattern, _, _) -> pattern) args_list |> pat_args
+        in
+        let _apply =
+          match rpc_type with
+          | `Connected_rpc_o ->
+              ( Asttypes.Nolabel
+              , [%expr Os_current_user.Opt.get_current_userid ()] )
+              :: List.map
+                   (fun (pattern, _, label) -> (label, pat_to_exp pattern))
+                   args_list
+          | _ ->
+              (Asttypes.Nolabel, [%expr Os_current_user.get_current_userid ()])
+              :: List.map
+                   (fun (pattern, _, label) -> (label, pat_to_exp pattern))
+                   args_list
+        in
+        ([ [%stri
+             let [%p pattern] =
+               [%expr
+                 fun [%p _args] ->
+                   [%e apply_function_expr ~loc ident.txt _apply]]]
+        ] [@metaloc loc])
     | _ -> (
         ([stri] [@metaloc loc]) )
+
+  (*let rpc_server rpc_type stri =
```

Comment:
I don't think this commented code is really needed.
If it is in a previous commit you can just get rid of it.